Scripts/Process/Moment/MomentProcess.cs

```diff
@@ -17,10 +17,71 @@ public sealed class MomentProcess : MonoBehaviour
         /// </summary>
         [Tooltip("Only run the process if the process is on an active GameObject and the Component is enabled.")]
         public bool onlyProcessOnActiveAndEnabled = true;
+        /// <summary>
+        /// A percentage defining how often to process the <see cref="process"/>.
+        /// </summary>
+        [Tooltip("A percentage defining how often to process the process."), Range(0f, 1f), SerializeField]
+        private float utilization = 1f;
+
+        /// <summary>
+        /// A percentage defining how often to process the <see cref="process"/>.
+        /// </summary>
+        public float Utilization
+        {
+            get
+            {
+                return utilization;
+            }
+            set
+            {
+                utilization = Mathf.Clamp(value, 0f, 1f);
+                UpdateDelay();
+            }
+        }
 
-        private void OnEnable()
+        /// <summary>
+        /// Keeps track of how often calls to <see cref="Process"/> were ignored because of <see cref="Utilization"/>.
+        /// </summary>
+        protected int counter;
+        /// <summary>
+        /// How many calls to <see cref="Process"/> to ignore because of <see cref="Utilization"/>.
+        /// </summary>
+        protected int delay;
+
+        /// <summary>
+        /// Calls <see cref="IProcessable.Process"/> on <see cref="process"/> if <see cref="Utilization"/> allows.
+        /// </summary>
+        public virtual void Process()
+        {
+            if (Utilization < float.Epsilon)
+            {
+                return;
+            }
+
+            if (process != null && (!onlyProcessOnActiveAndEnabled || isActiveAndEnabled) && counter == delay)

```

Comment:
The exception hasn't been carried over in case the interface process is null. Should we be doing that here too?

Reply:
I removed it as accessing it will give the same null ref exception and Unity logs the member name these days.